### PR TITLE
Add a sample containing MI and Ballerina code

### PR DESCRIPTION
**Pull request by poorna2152**
## Purpose
$subject

### CODE REVIEW

File: examples/price-calculation/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+```xml
+<total xmlns="http://ws.apache.org/ns/synapse">{total}</total>
+```
+
+## Steps to Invoke the Sample
+
+Follow these steps to invoke the sample using the connector:
+
+1. The `bal-price-calculation` folder contains the ballerina code for the connector. Invoke the following command to generate the connector:
```
Comment (MaryamZi) [inline examples/price-calculation/README.md]:
```suggestion
1. The `bal-price-calculation` folder contains the Ballerina code for the connector. Invoke the following command to generate the connector:
```

Comment (MaryamZi) [inline examples/price-calculation/README.md]: So we need to navigate there, right? Shall we mention that explicitly?

File: examples/price-calculation/bal-price-calculation/main.bal [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import ballerinax/mi;
+
+@mi:ConnectorInfo {}
```
Comment (MaryamZi) [inline examples/price-calculation/bal-price-calculation/main.bal]:
```suggestion
@mi:ConnectorInfo
```

Comment (MaryamZi) [reply examples/price-calculation/bal-price-calculation/main.bal]: This works, right?

File: examples/price-calculation/mi-price-calculation/deployment/deployment.toml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+[server]
+hostname = "localhost"
+hot_deployment = false
+# offset  = 10
```
Comment (MaryamZi) [inline examples/price-calculation/mi-price-calculation/deployment/deployment.toml]: Can we remove the commented out stuff?

File: examples/price-calculation/mi-price-calculation/src/main/wso2mi/artifacts/apis/price-calculation.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+     <resource methods="POST" uri-template="/">
+        <inSequence>
+            <bal_price_calculation.calculateTotal>
+                <Result>total</Result>
+                <invoice>{$body/*}</invoice>
+            </bal_price_calculation.calculateTotal>
+            <payloadFactory media-type="xml" template-type="default">
+                <format>
+                    <total>
+                    $1</total>
```
Comment (MaryamZi) [inline examples/price-calculation/mi-price-calculation/src/main/wso2mi/artifacts/apis/price-calculation.xml]:
```suggestion
                    <total>$1</total>
```

File: examples/price-calculation/mi-price-calculation/src/main/wso2mi/artifacts/apis/price-calculation.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+                </format>
+                <args>
+                    <arg expression="$ctx:total" evaluator="xml" />
+                </args>
+            </payloadFactory>
+            <respond />
```
Comment (MaryamZi) [inline examples/price-calculation/mi-price-calculation/src/main/wso2mi/artifacts/apis/price-calculation.xml]:
```suggestion
            <respond/>
```

File: examples/price-calculation/README.md (hunk 1 above)
Comment (poorna2152) [reply examples/price-calculation/README.md]: We don't have to navigate to that folder, we can provide the path to that folder.  Navigating to that is not working now due to a bug. https://github.com/ballerina-platform/ballerina-library/issues/7370. I'll leave this as it is then?

File: examples/price-calculation/README.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+## Steps to Invoke the Sample
+
+Follow these steps to invoke the sample using the connector:
+
+1. The `bal-price-calculation` folder contains the Ballerina code for the connector. Invoke the following command to generate the connector:
+    ```bash
+    bal mi -i <ballerina-project>
+    ```
+
+2. A zip file of the connector will be generated. Add this zip file to the MI project inside the folder `mi-price-calculation` following the approach described [here](https://mi.docs.wso2.com/en/latest/develop/creating-artifacts/adding-connectors/).
```
Comment (MaryamZi) [inline examples/price-calculation/README.md]: zip should be ZIP I think.

File: examples/price-calculation/README.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+         <price>25</price>
+      </item>
+      <item>
+         <name>pencil</name>
+         <price>12</price>
+      </item>
+   </items>`'
+```
+
+Output:
```
Comment (MaryamZi) [inline examples/price-calculation/README.md]:
```suggestion
Output:

```

Let's add after L27 also.

File: examples/price-calculation/README.md (hunk 1 above)
Comment (MaryamZi) [reply examples/price-calculation/README.md]: Ah yes, missed the argument. This is okay.

**Review by gimantha — APPROVED** [review]

**Pull request merged**